app/src/main/kotlin/com/adesso/movee/scene/login/LoginViewModel.kt

```diff
@@ -30,13 +31,13 @@ class LoginViewModel @Inject constructor(
     }
 
     fun onLoginClick() {
-        uiScope.launch {
+        viewModelScope.launch {
             val username = username.value ?: return@launch
             val password = password.value ?: return@launch
 
             _loginInProgress.value = true
 
-            val loginResult = onBackgroundThread {
+            val loginResult = runOnViewModelScope {
```

Comment:
`runOnViewModelScope` shouldn't be needed or used. We should be able to call suspend functions directly from launch block. see: https://developer.android.com/kotlin/coroutines/coroutines-best-practices#main-safe

Reply:
If I understood your concern correctly, the function in question is indeed an extension function, as you also understood. The reason I'm using it this way is that I converted the existing onBackgroundThread extension function to runOnViewModelScope with a small refactor in order to preserve the existing structure. Therefore, I continue to use the old extension function by simply changing its name. You can take a look at the BaseAndroidViewModel function for this. If you sure that we need to change the structure as you wish. It's fine to me.

Reply:
I saw the old one actually and thought it should be better to remove it all together. However, you are right, we can keep it like this and make refactoring with another PR.

Reply:
It's resolved.